dbms/src/TableFunctions/TableFunctionMySQL.cpp

```diff
@@ -147,23 +99,25 @@ StoragePtr TableFunctionMySQL::executeImpl(const ASTPtr & ast_function, const Co
         << " AND TABLE_NAME = " << quote << table_name
         << " ORDER BY ORDINAL_POSITION";
 
-    MySQLBlockInputStream result(pool.Get(), query.str(), sample_block, DEFAULT_BLOCK_SIZE);
-
     NamesAndTypesList columns;
+    MySQLBlockInputStream result(pool.Get(), query.str(), sample_block, DEFAULT_BLOCK_SIZE);
     while (Block block = result.read())
     {
         size_t rows = block.rows();
         for (size_t i = 0; i < rows; ++i)
             columns.emplace_back(
                 (*block.getByPosition(0).column)[i].safeGet<String>(),
-                getDataType(
+                convertMySQLDataType(
                     (*block.getByPosition(1).column)[i].safeGet<String>(),
                     (*block.getByPosition(2).column)[i].safeGet<UInt64>() && context.getSettings().external_table_functions_use_nulls,
                     (*block.getByPosition(3).column)[i].safeGet<UInt64>(),
                     (*block.getByPosition(4).column)[i].safeGet<UInt64>()));
 
     }
 
+    if (columns.empty())
+        throw Exception("MySQL table `" + database_name + "`.`" + table_name + "` doesn't exist.", ErrorCodes::UNKNOWN_TABLE);
```

Comment:
But there is a function named `backQuoteIfNeed` that will also do proper escaping.